src/server/status/Status.js

```diff
@@ -36,6 +36,22 @@ class Status extends EventEmitter {
       since: this.since
     };
   }
+
+  on(eventName, handler) {
+    super.on(eventName, handler);
+
+    if (eventName === this.state) {
+      setImmediate(handler);
```

Comment:
the context and arguments used when calling the handler need to match the [the original call site](https://github.com/w33ble/kibana/blob/status-listener-fire/src/server/status/Status.js#L78)